pkg/p2p/kadcast/bucket.go

```diff
@@ -66,27 +55,16 @@ func (b *bucket) removePeerAtIndex(index int) []Peer {
 	return b.entries[:len(b.entries)-1]
 }
 
-// Picks a random Peer from the bucket and returns it.
-func (b bucket) getRandomPeer() (*Peer, error) {
-	if b.peerCount == 0 {
-		// If the bucket has no nodes inside, we return an error.
-		return nil, errors.New("bucket empty")
-	}
-	maxVal := len(b.entries)
-	return &b.entries[rand.Intn(maxVal)], nil
-}
```

Comment:
Why has this function been deleted? I haven't seen a replacement for it and is needed to decide which nodes should I forward a `CHUNKS` message according to the `beta` parameter of the paper.

Reply:
The replacement of it is in `utils.go` (`generateRandomDelegates`). Also there is a unit test TestGenerateRandomDelegates.

`getRandomPeer` did not use beta value.

Reply:
Yep, I was calling the function `beta` times but it can drive to errors selecting the same target multiple times.

Good job!